components/price/price.js

```diff
@@ -0,0 +1,114 @@
+// Imports
+import React from 'react';
+import { getClassNamesWithMods } from '../_helpers';
+
+const { PropTypes } = React;
+
+/**
+ * Price component
+ */
+function Price({
+  dataAttrs = {},
+  decimalsPrecision,
+  decimalsSeparator,
+  mods = [],
+  symbol,
+  symbolPosition,
+  thousandsSeparator,
+  underlined,
+  value,
+}) {
+  const rootClass = 'ui-price';
+  const className = getClassNamesWithMods(rootClass, mods);
+  const [intValue, decValue] = value.toString().split('.');
+
+  let formattedIntValue = intValue;
+  if (thousandsSeparator && (formattedIntValue.length > 3)) {
+    formattedIntValue = formattedIntValue
+      .split('')
+      .reverse()
+      .reduce((ret, digit, idx) => ret.concat([((idx > 0) && ((idx % 3) === 0) ? thousandsSeparator : ''), digit]), [])
```

Comment:
what do you think about extract this function? for example, name it 'removeThousandsSeparator'. IMHO would be better for readability.